language_service/src/lib.rs

```diff
@@ -218,6 +219,17 @@ impl LanguageService {
         )
     }
 
+    /// LSP: textDocument/format
+    #[must_use]
+    pub fn get_format_changes(&self, uri: &str) -> Vec<TextEdit> {
+        self.document_op(
+            format::get_format_changes,
+            "get_format_changes",
+            uri,
+            Position { line: 0, column: 0 },
```

Comment:
Since position will never be a relevant parameter, I suggest taking the parameter out of this `get_format_changes` altogether. Here you can just pass in a lambda instead of the name of the function. Then to reduce confusion further you can pass `()` instead of `Position { line: 0, column: 0 }`. 

An example to copy is `get_code_lenses` in this file.

Reply:
It would be relevant when we add the functionality to format highlighted sections of code, but since that feature is not implemented yet, it doesn't make sense to leave the parameter in. Removed.